### PR TITLE
GOLD-211: ECDSA Multi-sig feature

### PR DESCRIPTION
**Pull request by arhamj**
Task: https://linear.app/shm/issue/GOLD-211/

- Added config to hold a mapping of allowed keys for multi-sig
- Adding and unifying multi-sig related flags.
- Adding support to accept a `verifyMultiSigs` method implementation from the app layer. To let curve specific logic to reside on the app layer.
- Updated and renamed middleware to use the above.
- Updated /df endpoint to use this middleware by default

### CODE REVIEW

File: src/debug/config.ts [hunk 1]
```diff
@@ -23,46 +23,45 @@ export function isServiceMode(): boolean {
...
-  if (pkClearance !== undefined) return key
-  return null
+  return getDevPublicKeys()[key] !== undefined ? key : null
 }
 
 export function getDevPublicKeyMaxLevel(clearance?: DevSecurityLevel): string | null {
   const devPublicKeys = getDevPublicKeys()
   let maxLevel = -Infinity
   let maxKey = null
-  for (const key in devPublicKeys) {
```
Comment (arhamj) [inline src/debug/config.ts]: Revert this

Comment (arhamj) [reply src/debug/config.ts]: And rest in this file

Comment (jintukumardas) [reply src/debug/config.ts]: reverted

**Review by afostr — APPROVED** [review]
Looks great.   local network smoke test passed.

**Pull request merged**